### PR TITLE
chore: Fix typos

### PR DESCRIPTION
**Pull request by NathanBaulch**
Just thought I'd contribute some typo fixes that I stumbled on. Nothing controversial (hopefully).

Use the following command to get a quick and dirty summary of the specific corrections made:
```shell
git diff HEAD^! --word-diff-regex='\w+' -U0 \
  | grep -E '\[\-.*\-\]\{\+.*\+\}' \
  | sed -r 's/.*\[\-(.*)\-\]\{\+(.*)\+\}.*/\1 \2/' \
  | sort | uniq -c | sort -n
```

FWIW, the top typos are:
* nsted (20)
* emtpy (13)
* detatch (6)
* unparseable (4)
* specficied (2)
* straming (2)
* inteegration (1)
* mutliple (1)
* modifer (1)

### CODE REVIEW

File: integration/meta-typings-as-const/google/protobuf/descriptor.ts [hunk 1]
```diff
@@ -167,7 +167,7 @@ export interface FieldDescriptorProto {
    * If true, this is a proto3 "optional". When a proto3 field is optional, it
    * tracks presence regardless of field type.
    *
-   * When proto3_optional is true, this field must be belong to a oneof to
```
Comment (stephenh) [inline integration/meta-typings-as-const/google/protobuf/descriptor.ts:170]: @NathanBaulch these `google/*/*.ts` files/comments come from comments within the google `*.proto` files that we can't change, so I think we need to revert any `descriptor.ts` changes.

File: protos/google/protobuf/compiler/plugin.proto [hunk 2]
```diff
@@ -103,7 +103,7 @@ message CodeGeneratorResponse {
   // This should be used to indicate errors in .proto files which prevent the
   // code generator from generating correct code.  Errors which indicate a
   // problem in protoc itself -- such as the input CodeGeneratorRequest being
-  // unparseable -- should be reported by writing a message to stderr and
+  // unparsable -- should be reported by writing a message to stderr and
```
Comment (stephenh) [inline protos/google/protobuf/compiler/plugin.proto:106]: These comments are also probably coming from a google `*.proto` file

**Review by stephenh — APPROVED** [review]
Hi @NathanBaulch , thanks for the PR! The typo fixes are appreciated. Just one ask to revert some of the comments in code generated from `*.proto` files that we don't control/can't fix the source typo in, but otherwise this is great.

**Comment by NathanBaulch** [comment]
Restored all typos in `**/google/protobuf/**/*.*` files, as requested.

File: integration/meta-typings-as-const/google/protobuf/descriptor.ts (hunk 1 above)
Comment (NathanBaulch) [reply integration/meta-typings-as-const/google/protobuf/descriptor.ts]: Done!

File: protos/google/protobuf/compiler/plugin.proto (hunk 2 above)
Comment (NathanBaulch) [reply protos/google/protobuf/compiler/plugin.proto]: Also done!

**Comment by stephenh** [comment]
Looks great, thanks @NathanBaulch !

**Pull request merged**